### ISSUE TITLE
"Bad wheel filename" if setuptools_scm is used

### ISSUE THREAD

**Issue by KOLANICH** [pypa/wheel#268]
The investigation revealed that WHEEL_INFO_RE has no group for commit hash.
The example of a failed build: https://gitlab.com/KOLANICH/AutoXGBoost.py/-/jobs/107657168

**Comment by agronholm** [comment]
Already fixed in master by c84867fae8eff746ab3380a2d5cf5fb2de3266dc.

**Comment by KOLANICH** [comment]
Still fa8ling: https://gitlab.com/KOLANICH/AutoXGBoost.py/-/jobs/108017296

**Comment by agronholm** [comment]
Hm, there seems to be a deeper conflict. According to the [wheel specification](https://www.python.org/dev/peps/pep-0427/#id12), the file name should only have alphanumeric characters or underscores in each component of the wheel file name. What I did was change the regular expression to accept any such file name, but these wheels have a `+` in the version specifier. I'm not sure where this is going wrong.

@dholth Should `bdist_wheel` mangle the `+` into an underscore? That's how I interpret the specification.

**Comment by KOLANICH** [comment]
I guess we need a new PEP documenting the thing actually happening and implementing it in the lib. Not adding a plus into version specifier, but adding a separate group for a commit hash.

**Comment by josiahjohnston** [comment]
Just ran into this after updating pip to 18.1 with our dev versioning that looks like `v1.x.y+GITSHA+localmod`. This combination of mangling `+` to `_` and having a more permissive RE looks like it will address my use case.
I spent some time today trying to figure out if I needed to change our versioning scheme. [PEP 440](https://www.python.org/dev/peps/pep-0440/#adding-local-version-identifiers) seemed to say the `+` was preferable over other characters for local version identifiers.

It would be nice if PEP 440 & 427 were compatible with each other, or at had enough consistency for me to feel confident about which versioning style will be supported.
